### PR TITLE
Fix `implicit-function-declaration` warnings, pass 5

### PR DESCRIPTION
**Pull request by ribbanya**
Finished `ft/chara` :rocket:

### CODE REVIEW

File: src/melee/ft/chara/ftZelda/ftzelda3.c [hunk 1]
```diff
@@ -84,46 +79,144 @@ void ftZelda_SpecialAirN_StartAction(HSD_GObj* fighter_gobj)
     fp->cb.x21BC_callback_Accessory4 = &ftZelda_8013A8AC;
 }
 
-// 8013AA38 - 8013AACC (0x94 bytes)
-// https://decomp.me/scratch/XcH4E
+#ifdef MUST_MATCH
+
+#pragma push
+asm void ftZelda_8013AA38(HSD_GObj* fighter_gobj)
```
Comment (ribbanya) [inline src/melee/ft/chara/ftZelda/ftzelda3.c:85]: Needed to revert a few functions in this file after correcting the type of `ftZeldaAttributes::x84`. It's probably not that hard to match if you want to try.

File: src/melee/ft/fighter.c [hunk 2]
```diff
@@ -144,7 +150,7 @@ void Fighter_FirstInitialize_80067A84()
 void Fighter_LoadCommonData(void)
 {
     void** pData;
-    func_80016C64("PlCo.dat", &pData, "ftLoadCommonData", 0);
+    func_80016C64("PlCo.dat", (unk_t) &pData, "ftLoadCommonData", 0);
```
Comment (ribbanya) [inline src/melee/ft/fighter.c:153]: Temporary hack but `fighter.c` is next up anyway.

File: src/melee/ft/fighter.h [hunk 3]
```diff
@@ -994,28 +995,6 @@ typedef struct _CollData {
...
-    unsigned char x24_5 : 1;  // 0x24 0x08
-    unsigned char x24_6 : 1;  // 0x24 0x04
-    unsigned char x24_7 : 1;  // 0x24 0x02
-    unsigned char x24_8 : 1;  // 0x24 0x01
-    Vec3 x28_hurt1_pos;       // 0x28
-    Vec3 x34_hurt2_pos;       // 0x34
-    u32 x40_bone_index;       // 0x40
-    u32 x44_hurt_kind;        // 0x44. 0 = low, 1 = mid, 2 = high
-    u32 x48_is_grabbable;     // 0x48
-} ftHurt;
```
Comment (ribbanya) [inline src/melee/ft/fighter.h:1017]: Moved to `lbcollision.h`, pending a rename.

File: src/melee/ft/chara/ftPikachu/ftpikachu3.c [hunk 4]
```diff
@@ -55,12 +57,14 @@ void ftPikachu_80124FA0(HSD_GObj* fighter_gobj)
     }
 }
 
-void ftPikachu_Stub_80124FDC()
+void ftPikachu_Stub_80124FDC(void)
 {
+    return;
```
[the remark this answers is not in the document]
Comment (ribbanya) [reply src/melee/ft/chara/ftPikachu/ftpikachu3.c:62]: Ok but I hate how
```c
void func()
{
}
```
looks. Can you also make `clang-format` do this first:
```c
void func() { }
```

File: src/melee/ft/chara/ftZelda/ftzelda3.c [hunk 5]
```diff
@@ -237,15 +428,17 @@ void ftZelda_8013AD1C(HSD_GObj* fighter_gobj)
                                        fp->x894_currentAnimFrame, 1.0, 0);
 
     fighter2 = fighter_gobj->user_data;
-    attributes = ((u32) fighter2->x2D4_specialAttributes);
+    attributes = fighter2->x2D4_specialAttributes;
 
     if (fighter2->x2200_ftcmd_var0 == 2U) {
-        ftColl_CreateReflectHit(fighter_gobj, attributes + 0x84,
+        ftColl_CreateReflectHit(fighter_gobj, attributes->x84,
```
[the remark this answers is not in the document]
Comment (ribbanya) [reply src/melee/ft/chara/ftZelda/ftzelda3.c:434]: Oh, that makes sense.

[the remark this answers is not in the document]
Comment (ribbanya) [reply src/melee/ft/chara/ftZelda/ftzelda3.c:434]: Doesn't seem as simple as just flipping the type (causes `stage.c` to mismatch in the data section). Can you look at it?
https://github.com/ribbanya/melee/commit/4ead5869aee365fd08c0d2e72e07b3a8ae18ab02

[the remark this answers is not in the document]
Comment (r-burns) [reply src/melee/ft/chara/ftZelda/ftzelda3.c:434]: Yeah just remove the other members from x88 onward

File: src/melee/ft/chara/ftPikachu/ftpikachu3.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (r-burns) [reply src/melee/ft/chara/ftPikachu/ftpikachu3.c:62]: Yeah clang-format can do that, let me take a look when I get home

[the remark this answers is not in the document]
Comment (r-burns) [reply src/melee/ft/chara/ftPikachu/ftpikachu3.c:62]: The option is `AllowShortFunctionsOnASingleLine: Empty`

**Review by r-burns — APPROVED** [review]

**Pull request merged**